homeassistant/components/sensor/filesize.py

```diff
@@ -0,0 +1,94 @@
+"""
+Sensor for monitoring the size of a file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filesize/
+"""
+import datetime
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_FILE_PATHS = 'file_paths'
+ICON = 'mdi:file'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FILE_PATHS):
+        vol.All(cv.ensure_list, [cv.isfile]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the file size sensor."""
+    sensors = []
+    for path in config.get(CONF_FILE_PATHS):
+        sensors.append(Filesize(path))
```

Comment:
All paths have to be whitelisted by checking it with `hass.config.is_allowed_path(path)`

Reply:
Done